### PR TITLE
Fix reference random numbers for CI

### PR DESCRIPTION
**Pull request by trossi**
This PR will
* Update reference random numbers to match those obtained with CPU execution (unreproducible on LUMI-G)
* Remove seeding from `updateZ.py` and do it only in test
* Enable CI on all branches and add pip caching

@gtikhonov Could you review?

### CODE REVIEW

File: requirements_dev.txt [hunk 1]
```diff
@@ -1,142 +1,8 @@
...
-jupyter_server_terminals==0.4.4
-jupyterlab==4.0.2
-jupyterlab-pygments==0.2.2
-jupyterlab_server==2.23.0
-keras==2.12.0
-kiwisolver==1.4.4
-libclang==16.0.0
-Markdown==3.4.3
-MarkupSafe==2.1.2
 matplotlib==3.7.1
```
Comment (gtikhonov) [inline requirements_dev.txt:1]: Are we plotting anything?

Comment (trossi) [reply requirements_dev.txt:1]: Only in [notebooks](https://github.com/aniskhan25/hmsc-hpc/tree/main/notebooks).

Comment (gtikhonov) [reply requirements_dev.txt:1]: Ok, these were useful but turned out to be unused (at this stage at least), so shall be moved to outside of the package. Therefore, the dependence shall be removed as well.

**Review by gtikhonov — APPROVED** [review]
Everything seems consistent to visual inspection.

**Pull request merged**